### ISSUE TITLE
Known accessibility issues

### ISSUE THREAD

**Issue by bjankord** [cerner/terra-ui#46]
# Issue Description
There are some accessibility issues in the doc site mainly related to heading order and color contrast of `<code>` blocks in markdown rendered content.

This issue serves as a collection of issues from manual testing with http://squizlabs.github.io/HTML_CodeSniffer/

## Issue Type

- [ ] New Feature
- [ ] Enhancement
- [x] Bug
- [ ] Other

## Expected Behavior
Doc site renders without any WCAG 2 AA or Section 508 errors reported from http://squizlabs.github.io/HTML_CodeSniffer/

### https://terra-ui.herokuapp.com/#/components/terra-alert/alert/alert
- [x] This form does not contain a submit button
- [x] This textinput element does not have a name available to an accessibility API
- [x] This form field should be labelled in some way
### https://terra-ui.herokuapp.com/#/components/terra-application-layout/application-layout/about
- [x] Duplicate id attribute value "is-required-false" found on the web page.
- [x] Duplicate id attribute value "is-required-true" found on the web page.
### https://terra-ui.herokuapp.com/#/components/terra-application-links/application-links/application-links
- [ ] Color contrast issues
### https://terra-ui.herokuapp.com/#/components/terra-application-utility/application-utility/application-utility
- [ ] Color contrast issue
- [x] Duplicate id attribute value "default" found on the web page.
### https://terra-ui.herokuapp.com/#/components/terra-arrange/arrange/arrange
- [x] This form does not contain a submit button
### https://terra-ui.herokuapp.com/#/components/terra-avatar/avatar/avatar
- [x] Color contrast issues
### https://terra-ui.herokuapp.com/#/components/terra-base/base/base
- [x] This link points to a named anchor "license" within the document, but no anchor exists with that name.
### https://terra-ui.herokuapp.com/#/components/terra-brand-footer/brand-footer/brand-footer
- [x] Anchor element found with a valid href attribute, but no link content has been supplied.
### https://terra-ui.herokuapp.com/#/components/terra-clinical-item-collection/clinical-item-collection
- [ ] Color contrast issues (Should be resolved with https://github.com/cerner/terra-clinical/issues/378)
- [x] The heading structure is not logically nested.
### https://terra-ui.herokuapp.com/#/components/terra-clinical-onset-picker/clinical-onset-picker
- [ ] This fieldset element does not have a name available to an accessibility API. 
- [ ] Fieldset does not contain a legend element.
- [ ] This select element does not have a name available to an accessibility API. 
- [ ] This form field should be labelled in some way.
There is currently no way to apply a legend for the field sets in this component. New issue logged here: https://github.com/cerner/terra-clinical/issues/435
### https://terra-ui.herokuapp.com/#/components/terra-date-picker/date-picker/date-picker
- [x] This textinput element does not have a name available to an accessibility API. 
- [x] This form field should be labelled in some way.
### https://terra-ui.herokuapp.com/#/components/terra-date-time-picker/date-time-picker/date-time-picker
- [x] This textinput element does not have a name available to an accessibility API. 
- [x] This form field should be labelled in some way.
### https://terra-ui.herokuapp.com/#/components/terra-dialog-modal/dialog-modal/dialog-modal
- [x] Duplicate id attribute value "trigger-dialog-modal"
### https://terra-ui.herokuapp.com/#/components/terra-dynamic-grid/dynamic-grid/dynamic-grid
- [x] Color contrast issues
### https://terra-ui.herokuapp.com/#/components/terra-embedded-content-consumer/embedded-content-consumer/embedded-content-consumer
- [x] Iframe element requires a non-empty title attribute that identifies the frame.
### https://terra-ui.herokuapp.com/#/components/terra-form-fieldset/form-fieldset/form-fieldset
- [x] This form does not contain a submit button
- [x] This textinput element does not have a name available to an accessibility API
- [x] This form field should be labelled in some way
### https://terra-ui.herokuapp.com/#/components/terra-form-input/form-input/input
- [x] This textinput element does not have a name available to an accessibility API
- [x] This form field should be labelled in some way.
- [x] The heading structure is not logically nested
### https://terra-ui.herokuapp.com/#/components/terra-form-select/form-select/select
- [ ] Color contrast issue (Should be resolved with https://github.com/cerner/terra-core/issues/1917)
### https://terra-ui.herokuapp.com/#/components/terra-form-textarea/form-textarea/textarea
- [x] This textarea element does not have a name available to an accessibility API
- [x] This form field should be labelled in some way.
### https://terra-ui.herokuapp.com/#/components/terra-form-validation/form-validation/validation-events
- [ ] Color contrast issue (Should be resolved with https://github.com/cerner/terra-core/issues/1917)
### https://terra-ui.herokuapp.com/#/components/terra-hookshot/hookshot/hookshot
- [x] This form does not contain a submit button
- [x] This numberinput element does not have a name available to an accessibility API
- [x] Duplicate id attribute value "hookshotAttachmentBehaviorundefined" 
- [x] Duplicate id attribute value "hookshotAttachmentMarginundefined" 
- [x] Duplicate id attribute value "hookshotContentAttachmentundefined" 
- [x] Duplicate id attribute value "hookshotTargetAttachmentundefined" 
### https://terra-ui.herokuapp.com/#/components/terra-i-18-n/i-18-n/i-18-n
- [x] This select element does not have a name available to an accessibility API
- [x] Select is missing label
- [x] The heading structure is not logically nested.
### https://terra-ui.herokuapp.com/#/components/terra-menu/menu/menu
- [x] This form does not contain a submit button
### https://terra-ui.herokuapp.com/#/components/terra-modal-manager/modal-manager/example
- [x] This form does not contain a submit button.
### https://terra-ui.herokuapp.com/#/components/terra-navigation-layout/navigation-layout/navigation-layout
- [x] The heading structure is not logically nested.
### https://terra-ui.herokuapp.com/#/components/terra-notification-dialog/notification-dialog/notification-dialog
- [x] The heading structure is not logically nested.
### https://terra-ui.herokuapp.com/#/components/terra-popup/popup/popup
- [x] This select element does not have a name available to an accessibility API
- [x] Select missing label
- [x]  Form does not contain a submit button
- [x] The heading structure is not logically nested.
### https://terra-ui.herokuapp.com/#/components/terra-section-header/section-header/section-header
- [x] The heading structure is not logically nested. 
### https://terra-ui.herokuapp.com/#/components/terra-slide-panel-manager/slide-panel-manager/slide-panel-manager
- [x] Form does not contain a submit button
- [x] Duplicate id attribute value "disclosureHeightdimensions"
- [x] Duplicate id attribute value "disclosureWidthdimensions"
### https://terra-ui.herokuapp.com/#/components/terra-status-view/status-view/status-view
- [x] Form does not contain a submit button
### https://terra-ui.herokuapp.com/#/components/terra-table/table/table
- [x] Color contrast issue with sample content
### https://terra-ui.herokuapp.com/#/components/terra-tag/tag/tag
- [ ] Color contrast issue with component
### https://terra-ui.herokuapp.com/#/components/terra-theme-provider/theme-provider/theme-provider
- [x] Select is missing label
- [x] The heading structure is not logically nested.
### https://terra-ui.herokuapp.com/components/core/time-input
- [x] Inputs missing labels
### Common issues
- [x] Color contrast for code blocks
- [x] Duplicate ids for change log headers
- [x] Links are indicated as an anchor since they start with #.

## Steps to Reproduce

1. Navigate to URL
2. Run http://squizlabs.github.io/HTML_CodeSniffer/ bookmarklet
3. Check WCAG 2 AA errors
4. Check Section 508 errors

**Comment by jmsv6d** [comment]
Needs design input for color contrast on `<code>` blocks.

**Comment by bjankord** [comment]
Text in markdown blockquotes seems to render a bit lighter compared to regular text from a markdown file. We should probably set blockquote text to same color when rendered from markdown file.

<img width="882" alt="screen shot 2019-01-02 at 8 20 48 pm" src="https://user-images.githubusercontent.com/633148/50621522-0c0ff380-0ecc-11e9-8b3b-a0ed1a136953.png">

**Comment by bjankord** [comment]
Closing as complete. Will log any issues we find from working on https://github.com/cerner/terra-ui/pull/141 to https://github.com/cerner/terra-ui/issues/138
